sql/core/src/main/scala/org/apache/spark/sql/catalog/interface.scala

```diff
@@ -98,23 +98,22 @@ class Table(
     val isTemporary: Boolean)
   extends DefinedByConstructorParams {
 
+  if (namespace != null) {
+    assert(namespace.forall(_ != null))
+  }
+
   def this(
       name: String,
       database: String,
       description: String,
       tableType: String,
       isTemporary: Boolean) = {
-    this(name, null, Array(database), description, tableType, isTemporary)
+    this(name, null, if (database != null) Array(database) else null,
```

Comment:
I don't think it's a serde problem, but about the semantic. When people using the old constructor and pass null as `database`, it means database is unknonw, so `namespace` should also be null instead of `Array(null)` to represent unknown.

Reply:
ok, let me clarify the title and description